### PR TITLE
dumpling, lightning: add document for compressed files support

### PR DESCRIPTION
**Pull request by lichunzhu**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Dumpling/Lightning have supported exporting/importing compressed files. This document add document about how to use them.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.3 (TiDB 6.3 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s): https://github.com/pingcap/tidb/issues/38514

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: dumpling-overview.md [hunk 1]
```diff
@@ -98,6 +98,14 @@ dumpling -u root -P 4000 -h 127.0.0.1 --filetype sql -t 8 -o /tmp/test -r 200000
 >
 > Dumpling 导出不区分*字符串*与*关键字*。如果导入的数据是 Boolean 类型的 `true` 和 `false`，导出时会被转换为 `1` 和 `0` 。
 
+### 压缩导出的数据文件
+
+你可以通过使用 `--compress gzip` 压缩导出的 csv/sql 数据与表结构文件。该参数支持 `gzip`、`snappy`、`zstd` 压缩算法，默认不压缩。
+
+- 该选项只能压缩每个数据/表结构文件，无法直接压缩整个文件夹生成单个压缩集合包。
+- 该选项可以节省磁盘空间，但也会导致导出速度变慢并增加 CPU 消耗，对导出速度要求较高的场景需慎用。
+- Lightning v6.5 及以上版本支持直接使用 dumpling 压缩文件作为数据源导入，无需额外配置。
```
Comment (hfxsd) [inline dumpling-overview.md:107]:
```suggestion
你可以使用 `--compress gzip` 压缩导出的 CSV、SQL 数据与表结构文件。该参数支持 `gzip`、`snappy`、`zstd` 压缩算法。默认不压缩。

- 该选项只能压缩每个数据与表结构文件，无法直接压缩整个文件夹生成单个压缩集合包。
- 该选项可以节省磁盘空间，但也会导致导出速度变慢并增加 CPU 消耗。对导出速度要求较高的场景需慎用。
- TiDB Lightning v6.5 及以上版本支持直接使用 Dumpling 压缩文件作为数据源导入，无需额外配置。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 2]
```diff
@@ -22,6 +22,7 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
 |Schema 文件|包含 `CREATE DATABASE` DDL 语句的文件|`${db_name}-schema-create.sql`|
 |数据文件|包含整张表的数据文件，该文件会被导入 `${db_name}.${table_name}` 表 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}</code>|
 |数据文件| 如果一个表分布于多个数据文件，这些文件命名需加上文件编号的后缀 | <code>\${db_name}.\${table_name}.001.\${csv\|sql\|parquet}</code> |
+|压缩文件| 上述所有类型文件如带压缩文件名后缀，lightning 会流式解压后进行导入 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}.{compress}</code> |
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:25]:
```suggestion
|压缩文件| 上述所有类型文件如带压缩文件名后缀，TiDB Lightning 会流式解压后进行导入 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}.{compress}</code> |
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 3]
```diff
@@ -283,6 +284,17 @@ type = '$3'
...
+## 压缩文件
+
+TiDB Lightning 目前支持由 Dumpling 导出的压缩文件或满足符合上文命名规则的压缩文件，目前支持 `gzip`, `snappy`, `zstd` 压缩算法的压缩文件。在文件名符合命名规则时，TiDB Lightning 会自动识别压缩算法在流式解压后导入，无需额外配置。
+
+注意：
+
+1. 由于 TiDB Lightning 无法对单个大压缩文件进行并发解压，因此压缩文件的大小会直接影响导入速度。建议压缩数据文件解压后的源文件大小不超过 256 MiB。
+2. TiDB Lightning 仅支持导入各自独立压缩的数据文件，不支持导入多个数据文件组成的单个压缩文件集合包。
+3. TiDB Lightning 不支持二次压缩的 `parquet` 文件，例如 `db.table.parquet.snappy`，如需压缩 `parquet` 文件用户可配置 `parquet` 文件数据存储的压缩格式。
+4. v6.4 及之后版本的 TiDB Lightning 会识别带有非 `.bak` 后缀的数据文件为 TiDB Lightning 不支持的压缩数据文件并报错，用户需要提前清理通过文件改名或将该类文件移出导入数据目录来避免此类错误。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:296]:
```suggestion
TiDB Lightning 目前支持由 Dumpling 导出的压缩文件或满足符合上文命名规则的压缩文件，目前支持 `gzip`、`snappy`、`zstd` 压缩算法的压缩文件。在文件名符合命名规则时，TiDB Lightning 会自动识别压缩算法在流式解压后导入，无需额外配置。

> **注意**
>
> - 由于 TiDB Lightning 无法对单个大压缩文件进行并发解压，因此压缩文件的大小会直接影响导入速度。建议压缩数据文件解压后的源文件大小不超过 256 MiB。
- TiDB Lightning 仅支持导入各自独立压缩的数据文件，不支持导入多个数据文件组成的单个压缩文件集合包。
- TiDB Lightning 不支持二次压缩的 `parquet` 文件，例如 `db.table.parquet.snappy`。如需压缩 `parquet` 文件，你可以配置 `parquet` 文件数据存储的压缩格式。
- TiDB v6.4 及之后版本的 TiDB Lightning 会不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。
```

File: tidb-lightning/troubleshoot-tidb-lightning.md [hunk 4]
```diff
@@ -210,3 +210,7 @@ header = false
 ### `Unknown character set`
 
 由于 TiDB 只支持部分 MySQL 字符集，因此，在导入流程中，如果创建表结构时使用了 TiDB 不支持的字符集，TiDB Lightning 会报这个错误。你可以结合数据内容选择 [TiDB 支持的字符集](/character-set-and-collation.md)，预先在下游创建表结构以绕过这个错误。
+
+### `invalid compression type ...`
+
+v6.4 及之后版本的 TiDB Lightning 会识别带有非 `.bak` 后缀的数据文件为 TiDB Lightning 不支持的压缩数据文件并报错，因此需要提前清理通过文件改名或将该类文件移出导入数据目录来避免此类错误。
```
Comment (hfxsd) [inline tidb-lightning/troubleshoot-tidb-lightning.md:216]:
```suggestion
TiDB v6.4 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。
```

File: dumpling-overview.md [hunk 5]
```diff
@@ -98,6 +98,14 @@ dumpling -u root -P 4000 -h 127.0.0.1 --filetype sql -t 8 -o /tmp/test -r 200000
 >
 > Dumpling 导出不区分*字符串*与*关键字*。如果导入的数据是 Boolean 类型的 `true` 和 `false`，导出时会被转换为 `1` 和 `0` 。
 
+### 压缩导出的数据文件
+
+你可以使用 `--compress gzip` 压缩导出的 CSV、SQL 数据与表结构文件。该参数支持 `gzip`、`snappy`、`zstd` 压缩算法。默认不压缩。
```
[the remark this answers is not in the document]
Comment (lichunzhu) [reply dumpling-overview.md:103]: SGTM，改成 `<format>` 比较好

Comment (hfxsd) [inline dumpling-overview.md:103]:
```suggestion
你可以使用 `--compress <format>` 压缩导出的 CSV、SQL 数据与表结构文件。该参数支持 `gzip`、`snappy`、`zstd` 压缩算法。默认不压缩。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 6]
```diff
@@ -22,6 +22,7 @@ TiDB Lightning 运行时将查找 `data-source-dir` 中所有符合命令规则
 |Schema 文件|包含 `CREATE DATABASE` DDL 语句的文件|`${db_name}-schema-create.sql`|
 |数据文件|包含整张表的数据文件，该文件会被导入 `${db_name}.${table_name}` 表 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}</code>|
 |数据文件| 如果一个表分布于多个数据文件，这些文件命名需加上文件编号的后缀 | <code>\${db_name}.\${table_name}.001.\${csv\|sql\|parquet}</code> |
+|压缩文件| 上述所有类型文件如带压缩文件名后缀，TiDB Lightning 会流式解压后进行导入 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}.{compress}</code> |
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:25]:
```suggestion
|压缩文件| 上述所有类型文件如带压缩文件名后缀，如 `gzip`、`snappy` 或 `zstd`，TiDB Lightning 会流式解压后进行导入 | <code>\${db_name}.\${table_name}.\${csv\|sql\|parquet}.{compress}</code> |
```

File: tidb-lightning/troubleshoot-tidb-lightning.md [hunk 7]
```diff
@@ -210,3 +210,7 @@ header = false
 ### `Unknown character set`
 
 由于 TiDB 只支持部分 MySQL 字符集，因此，在导入流程中，如果创建表结构时使用了 TiDB 不支持的字符集，TiDB Lightning 会报这个错误。你可以结合数据内容选择 [TiDB 支持的字符集](/character-set-and-collation.md)，预先在下游创建表结构以绕过这个错误。
+
+### `invalid compression type ...`
+
+TiDB v6.4 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。更多详情请参考[压缩导出的数据文件](/dumpling-overview.md#压缩导出的数据文件)。
```
Comment (hfxsd) [inline tidb-lightning/troubleshoot-tidb-lightning.md:216]:
```suggestion
TiDB v6.4 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。更多详情请参考[压缩导出的数据文件](/tidb-lightning/tidb-lightning-data-source.md#压缩文件)。
```

File: tidb-lightning/tidb-lightning-data-source.md [hunk 8]
```diff
@@ -283,6 +284,17 @@ type = '$3'
...
+## 压缩文件
+
+TiDB Lightning 目前支持由 Dumpling 导出的压缩文件或满足符合上文命名规则的压缩文件，目前支持 `gzip`、`snappy`、`zstd` 压缩算法的压缩文件。在文件名符合命名规则时，TiDB Lightning 会自动识别压缩算法在流式解压后导入，无需额外配置。
+
+> **注意**
+>
+> - 由于 TiDB Lightning 无法对单个大压缩文件进行并发解压，因此压缩文件的大小会直接影响导入速度。建议压缩数据文件解压后的源文件大小不超过 256 MiB。
+- TiDB Lightning 仅支持导入各自独立压缩的数据文件，不支持导入多个数据文件组成的单个压缩文件集合包。
+- TiDB Lightning 不支持二次压缩的 `parquet` 文件，例如 `db.table.parquet.snappy`。如需压缩 `parquet` 文件，你可以配置 `parquet` 文件数据存储的压缩格式。
+- TiDB v6.4 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。
```
Comment (hfxsd) [inline tidb-lightning/tidb-lightning-data-source.md:296]:
```suggestion
- TiDB v6.4 及之后版本的 TiDB Lightning 支持后缀为压缩后缀名 `gzip`、`snappy` 、 `zstd`，以及 `.bak` 的数据文件。其他后缀名会报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。
```

File: tidb-lightning/troubleshoot-tidb-lightning.md [hunk 9]
```diff
@@ -210,3 +210,7 @@ header = false
 ### `Unknown character set`
 
 由于 TiDB 只支持部分 MySQL 字符集，因此，在导入流程中，如果创建表结构时使用了 TiDB 不支持的字符集，TiDB Lightning 会报这个错误。你可以结合数据内容选择 [TiDB 支持的字符集](/character-set-and-collation.md)，预先在下游创建表结构以绕过这个错误。
+
+### `invalid compression type ...`
+
+TiDB v6.4 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。更多详情请参考[压缩导出的数据文件](/tidb-lightning/tidb-lightning-data-source.md#压缩文件)。
```
Comment (shichun-0415) [inline tidb-lightning/troubleshoot-tidb-lightning.md:216]:
```suggestion
TiDB v6.4.0 及之后版本的 TiDB Lightning 不支持带有非 `.bak` 后缀的数据文件并报错。你需要提前修改文件名，或将该类文件移出导入数据目录来避免此类错误。更多详情请参考[压缩导出的数据文件](/tidb-lightning/tidb-lightning-data-source.md#压缩文件)。
```

**Review by hfxsd — APPROVED** [review]

**Review by shichun-0415 — APPROVED** [review]
LGTM

**Pull request merged**